.github/workflows/build_and_test.yml

```diff
@@ -183,6 +183,12 @@ jobs:
       with:
         name: test-results-${{ matrix.modules }}-${{ matrix.comment }}-${{ matrix.java }}-${{ matrix.hadoop }}-${{ matrix.hive }}
         path: "**/target/test-reports/*.xml"
+    - name: Upload unit tests log files
+      if: failure()
+      uses: actions/upload-artifact@v2
+      with:
+        name: unit-tests-log-${{ matrix.modules }}-${{ matrix.comment }}-${{ matrix.java }}-${{ matrix.hadoop }}-${{ matrix.hive }}
```

Comment:
nit though, adding a prefix `jdk` for `matrix.java` will make the log name clearer just like `hadoop3.2` and `hive2.3`?

Reply:
Yup. I plan to have a proper name instead of combination of the matrix to address such issue clearly. I'll make a separate PR to address the issue of all such names in this whole file later.